### PR TITLE
Configurations/windows-makefile.tmpl: HTMLDOCS are files, not directories

### LINKED ISSUE #10553: OpenSSL failed to nmake clean due to fatal error U1095 on windows with MSVC
Environment:
Windows Server 2016 + VS2017 + openssl master branch latest source code.

Issue description:
I tried to build openssl using VS2017 on Windows Server 2016. We found that openssl failed to nmake clean due to fatal error U1095: expanded command line 'rmdir /Q /S doc\html\man3\ADMISSIONS.html  on windows with MSVC. It can be reproduced on latest master revision be3acd7 reversion. Could you please help take a look at this? Thanks in advance!

Repro steps:
1. git clone https://github.com/openssl/openssl D:\openssl\src
2. cd D:\openssl\src
3. perl Configure VC-WIN32
4. nmake clean

[log_x86_build_1.log](https://github.com/openssl/openssl/files/3909253/log_x86_build_1.log)

Actual result:
NMAKE : fatal error U1095: expanded command line 'rmdir /Q /S doc\html\man3\ADMISSIONS.html

### PR DESCRIPTION
**Pull request by levitte**
Remove them using "del", not "rmdir"

Fixes #10553

### CODE REVIEW

**Comment by mspncp** [comment]
It works, but it still takes a long time to configure and a terribly long time to clean.

Under Windows, it seems to have a great performance penalty to delete all files explicitly instead of doing a wilcard removal:

```
del /Q /F doc\html\man3\*.html
```

**Comment by levitte** [comment]
Yes, configuration takes long, because `doc/build.info` is generated dynamically.  I did propose that it could be generated with `make update`, but that got some pushback, so...

As for the rest, fixup coming up

File: Configurations/windows-makefile.tmpl [hunk 1]
```diff
@@ -402,10 +416,10 @@ libclean:
...
 
 clean: libclean
-	{- join("\n\t", map { "-del /Q /F $_" } @HTMLDOCS1) -}
-	{- join("\n\t", map { "-del /Q /F $_" } @HTMLDOCS3) -}
-	{- join("\n\t", map { "-del /Q /F $_" } @HTMLDOCS5) -}
-	{- join("\n\t", map { "-del /Q /F $_" } @HTMLDOCS7) -}
+	-rmdir /Q /S $(HTMLDOCS1_BLDDIRS)
+	-rmdir /Q /S $(HTMLDOCS3_BLDDIRS)
+	-rmdir /Q /S $(HTMLDOCS5_BLDDIRS)
+	-rmdir /Q /S $(HTMLDOCS7_BLDDIRS)
```
Comment (mspncp) [inline Configurations/windows-makefile.tmpl]: Something is wrong here; it seems like the variables expand to empty strings:

```diff
$ diff nmake-old.log nmake-new.log 
--- nmake-old.log	2019-12-02 12:10:28.353615678 +0100
+++ nmake-new.log	2019-12-02 12:08:56.490698028 +0100
@@ -9,28 +9,14 @@
 C:\src\openssl\providers\libcommon_static.lib konnte nicht gefunden werden
 C:\src\openssl\test\libtestutil_static.lib konnte nicht gefunden werden
 C:\src\openssl\libcrypto.* konnte nicht gefunden werden
-	del /Q /F doc\html\man1\CA.pl.html
-C:\src\openssl\doc\html\man1\CA.pl.html konnte nicht gefunden werden
-	del /Q /F doc\html\man1\openssl-asn1parse.html
-C:\src\openssl\doc\html\man1\openssl-asn1parse.html konnte nicht gefunden werden
-	del /Q /F doc\html\man1\openssl-ca.html
-C:\src\openssl\doc\html\man1\openssl-ca.html konnte nicht gefunden werden
-	del /Q /F doc\html\man1\openssl-ciphers.html
-C:\src\openssl\doc\html\man1\openssl-ciphers.html konnte nicht gefunden werden
-	del /Q /F doc\html\man1\openssl-cmds.html
-C:\src\openssl\doc\html\man1\openssl-cmds.html konnte nicht gefunden werden
-	del /Q /F doc\html\man1\openssl-cms.html
-C:\src\openssl\doc\html\man1\openssl-cms.html konnte nicht gefunden werden
-	del /Q /F doc\html\man1\openssl-crl.html
-
-...
-
-
-C:\src\openssl\doc\html\man7\proxy-certificates.html konnte nicht gefunden werden
-	del /Q /F doc\html\man7\ssl.html
-C:\src\openssl\doc\html\man7\ssl.html konnte nicht gefunden werden
-	del /Q /F doc\html\man7\x509.html
-C:\src\openssl\doc\html\man7\x509.html konnte nicht gefunden werden
+	rmdir /Q /S 
+Syntaxfehler.
+	rmdir /Q /S 
+Syntaxfehler.
+	rmdir /Q /S 
+Syntaxfehler.
+	rmdir /Q /S 
+Syntaxfehler.
 	del /Q /F apps\openssl.exe
 C:\src\openssl\apps\openssl.exe konnte nicht gefunden werden
 	del /Q /F fuzz\asn1-test.exe
```

Comment (levitte) [reply Configurations/windows-makefile.tmpl]: I just observed the exact same thing.  Debugging reveals that the build.info data is generated correctly, but still, nothing ends up in configdata.pm...

Comment (levitte) [reply Configurations/windows-makefile.tmpl]: Finally figured it out.  Stoopid error, and as usual, they take time to actually see.

Comment (mspncp) [reply Configurations/windows-makefile.tmpl]: I cleaned a full build and watched the performance. Your fix works and performance is much better than at the beginning of the pr. One more possible optimization would be to delete the test executables using a wild card.
```
 del /Q /F test\*.exe
``` 
Deleting every single one in a separate command is very slow. But you don't have to do it if you don't want to.

**Review by mspncp — APPROVED** [review]

**Comment by mspncp** [comment]
Apropos slow: doing a full build of OpenSSL on Windows is currently really painfully slow. It feels like build times have doubled since 1.1.1., because many modules are now built for the default and the FIPS provider.

**Comment by levitte** [comment]
Merged.

812500651954a7a084550d72d2f8ac9dc3158327 Configurations/windows-makefile.tmpl: HTMLDOCS are files, not directories

**Pull request closed without merging**